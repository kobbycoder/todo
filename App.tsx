import React from "react";
import { SafeAreaView, StyleSheet, Platform, StatusBar } from "react-native";
import Main from "./src/Main";

export default function App() {
  return (
    <SafeAreaView style={styles.AndroidSafeArea}>
      <Main />
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  AndroidSafeArea: {
    flex: 1,
    backgroundColor: "white",
    paddingTop: Platform.OS === "android" ? StatusBar.currentHeight : 0,
  },
});
