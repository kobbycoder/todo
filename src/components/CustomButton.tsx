import { StyleSheet, Text, TouchableWithoutFeedback } from "react-native";
import React from "react";
import { TouchableOpacity } from "react-native-gesture-handler";


interface CustomButtonProps {
    buttonText: string;
    onPress: () => void;
  }

const Buttons: React.FC<CustomButtonProps> = ({ buttonText, onPress }) => {

  return (
    <TouchableOpacity style={styles.button} onPress={onPress}>
      <Text style={styles.buttonText}>{buttonText}</Text>
    </TouchableOpacity>
  );
};

export default Buttons;

const styles = StyleSheet.create({
  button: {
    backgroundColor: "#001C30",
    padding: 12,
    borderRadius: 8,
    alignItems: "center",
    justifyContent: "center",
    width: "70%",
    marginVertical: 20,
  },
  buttonText: {
    color: "white",
    fontSize: 16,
    fontWeight: "bold",
  },
});
