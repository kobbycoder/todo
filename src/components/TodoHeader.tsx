import React from "react";
import { Image, StyleSheet, Text, View } from "react-native";

const TodoHeader = () => {

  return (
    <View style={[styles.header,styles.elevation, styles.shadowProp]}>
      <View>
        <Text style={styles.headerText}>To Do</Text>
      </View>
      <View>
        <Image source={require('../../assets/logo.png')}/>
      </View>
    </View>
  );
};

export default TodoHeader;

const styles = StyleSheet.create({
  header: {
    flexDirection: "row",
    width: "100%",
    height: "45%",
    backgroundColor: "#2dd4bf",
    alignItems: "center",
    paddingHorizontal: 20,
    justifyContent: "space-between",
  },
  headerText: {
    color: "white",
    fontSize: 20,
    fontFamily: "Poppins-Bold"
  },
  elevation: {
    elevation: 20,
    shadowColor: '#cccccc'
  },
  shadowProp: {
    shadowColor: '#171717',
    shadowOffset: {width: -2, height: 4},
    shadowOpacity: 0.2,
    shadowRadius: 3,
  },
});
