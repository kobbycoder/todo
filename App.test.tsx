import React from 'react';
import renderer from 'react-test-renderer';

import CustomButton from './src/components/CustomButton';
import TodoHeader from './src/components/TodoHeader';
import App from './App'

describe('<CustomButton />', () => {
  it('has 1 child', () => {
    const tree = renderer.create(<CustomButton />).toJSON();
    expect(tree?.children?.length).toBe(1);
  });
});

describe('<TodoHeader />', () => {
  it('has 2 child', () => {
    const tree = renderer.create(<TodoHeader />).toJSON();
    expect(tree?.children?.length).toBe(2);
  });
});

