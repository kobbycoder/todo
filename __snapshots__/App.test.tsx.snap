// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomButton /> renders correctly 1`] = `null`;

exports[`<TodoHeader /> renders correctly 1`] = `null`;
